frontend/src/Components/RoadDetails/ImageGallery.tsx

```diff
@@ -72,8 +73,14 @@ const ImageGallery: React.FC = () => {
   const [scrollPosition, setScrollPosition] = useState(0);
   const galleryRef = useRef<HTMLDivElement>(null);
 
-  const openImageInNewTab = (imageUrl: string) => {
-    window.open(imageUrl, '_blank');
+  // Add feature for the pop-up of clicking and checking images
+  const [isImageClickOpen, setIsImageClickOpen] = useState(false);
+  const [currentImageIndex, setCurrentImageIndex] = useState(0);
+
+  // Function to open the pop-up
+  const openImageInPopup = (imageId: number) => {
+    setCurrentImageIndex(imageId - 1); // Adjust for 0-based index
```

Comment:
```suggestion
  const openImageInPopup = (imageIdx: number) => {
    setOpenImageUrl(image[imageIdx]); 
```
Two things here:
- Changing from `setCurrentImageIndex`  to `setOpenImageUrl`
- use image index instead of image id (the image id will probably come from the back end and won't be as nice as right now)